amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpSinkStage.scala

```diff
@@ -77,3 +77,69 @@ final class AmqpSinkStage(settings: AmqpSinkSettings)
 
   override def toString: String = "AmqpSink"
 }
+
+
+object AmqpReplyToStage {
+
+  /**
+    * Internal API
+    */
+  private val defaultAttributes =
+    Attributes.name("AmqpReplyToSink").and(ActorAttributes.dispatcher("akka.stream.default-blocking-io-dispatcher"))
+}
+
+/**
+  * Connects to an AMQP server upon materialization and sends incoming messages to the server.
+  * Each materialized sink will create one connection to the broker. This stage sends messages to
+  * the queue named in the replyTo options of the message instead of from settings declared at construction.
+  */
+final class AmqpReplyToStage(settings: AmqpSinkSettings)
+extends GraphStage[SinkShape[OutgoingMessage]]
+with AmqpConnector { stage =>
+  import AmqpReplyToStage._
+
+  val in = Inlet[OutgoingMessage]("AmqpReplyToSink.in")
+
+  override def shape: SinkShape[OutgoingMessage] = SinkShape.of(in)
+
+  override protected def initialAttributes: Attributes = defaultAttributes
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) with AmqpConnectorLogic {
+      override val settings = stage.settings
+
+      override def connectionFactoryFrom(settings: AmqpConnectionSettings) = stage.connectionFactoryFrom(settings)
+
+      override def whenConnected(): Unit = {
+        val shutdownCallback = getAsyncCallback[ShutdownSignalException] { ex =>
+          failStage(ex)
+        }
+        channel.addShutdownListener(new ShutdownListener {
+          override def shutdownCompleted(cause: ShutdownSignalException): Unit =
+            shutdownCallback.invoke(cause)
+        })
+        pull(in)
+      }
+
+      setHandler(in,
+        new InHandler {
+          override def onPush(): Unit = {
+            val elem = grab(in)
+            channel.basicPublish(
+              "",
+              elem.props.map(_.getReplyTo).get,
```

Comment:
I'm open to opinions. I think the best would be to remove the ability to configure a destination in the settings. I suppose it could be an option for failing the stage or discarding the element?